lib/tapioca/loaders/loader.rb

```diff
@@ -79,15 +97,39 @@ def load_rails_engines
         end
       end
 
-      sig { returns(T::Array[T.untyped]) }
+      sig { returns(T.untyped) }
+      def rails_autoloader
+        return @rails_autoloader if defined?(@rails_autoloader)
+
+        @rails_autoloader = T.let(Rails.autoloaders.once, T.untyped) if Rails.respond_to?(:autoloaders)
+      end
+
+      sig { params(blk: T.proc.void).void }
+      def with_rails_application(&blk)
+        # Store the current Rails.application object so that we can restore it
+        rails_application = Rails.application
+
+        # Create a new Rails::Application object, so that we can load the engines.
+        # Some engines and the `Rails.autoloaders` call might expect `Rails.application`
+        # to be set, so we need to create one here.
+        unless rails_application
+          Rails.application = Class.new(Rails::Application)
+        end
+
+        blk.call
+      ensure
+        Rails.app_class = Rails.application = rails_application
+      end
+
+      T::Sig::WithoutRuntime.sig { returns(T::Array[T.class_of(Rails::Engine)]) }
       def rails_engines
         return [] unless Object.const_defined?("Rails::Engine")
```

Comment:
I think we can make this:
```suggestion
        return [] unless defined?(Rails::Engine)
```
which reads better than `const_defined?` and the constant reference can be typechecked.